test/asynchronous/test_session.py

```diff
@@ -394,12 +394,22 @@ async def test_cursor_clone(self):
         await cursor.close()
         await clone.close()
 
+    async def test_bind_session(self):
+        coll = self.client.pymongo_test.collection
+
         # Explicit session via context variable.
         async with self.client.start_session(bind=True) as s:
             cursor = coll.find()
             self.assertTrue(cursor.session is s)
-            clone = cursor.clone()
-            self.assertTrue(clone.session is s)
+
+        # Nested sessions.
+        session1 = self.client.start_session(bind=True)
+        with session1:
+            session2 = self.client.start_session(bind=True)
+            with session2:
+                coll.find_one()  # uses session2
+            coll.find_one()  # uses session1
+        coll.find_one()  # uses implicit session
```

Comment:
This test has to actually verify the correct sessions are used via command monitoring events.

Reply:
@ShaneHarvey I'm assuming that commit is wrong, I should be using command monitoring via event listeners right?